.github/workflows/docs-build.yml

```
@@ -0,0 +1,62 @@
+name: docs-build
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out mitiq
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          make install requirements
```

IIRC this should do the same thing. Can you double check? Maybe in the past it was a different command, but now I think `make install xyz` performs the same action as `make install`.

```suggestion
          make install
```

---

I do not know how a makefile works that well. It's on my list of todos.  

> Maybe in the past it was a different command,

Looking through the targets, were we using `make install install` or `make install requirements` in the past? According to [this](https://superuser.com/a/360180), `make install` will install from the first installation target which is `install`. 

https://github.com/unitaryfund/mitiq/blob/231319df9051dfbc61db79d02bee201979528a36/Makefile#L49-L51

https://github.com/unitaryfund/mitiq/blob/231319df9051dfbc61db79d02bee201979528a36/Makefile#L59-L61

I see `development` is utilized in `setup.py` where we install things listed in `requirements/`. 

Why do we have two separate installation targets in the makefile? IMO both appear to behave the same, unless I am missing something.

---

> `pip install -e .[development]`

This will install
1. an editable installation of Mitiq (`-e` is short for `--editable`)
2. **all** the dependencies (the `[development]` "tag" is defined in `setup.py` [here](https://github.com/unitaryfund/mitiq/blob/02ebc4bf5291b1d8abfec7516dd3bc5a9911cc37/setup.py#L37-L40) to mean all of the dependencies found in `requirements/`.)

> `pip install -r requirements/requirements.txt`

This will _only_ install the requirements listed in `requirements/requirements.txt` which is currently `numpy`, `scipy`, `cirq-core`, and `tabulate`. I can't find any uses of this command in Mitiq and it should be safe to remove. I've added a note for the community call this week to discuss.

---

Discussed in the community call. Green light from everyone on removing it. 
@purva-thakre is going to create a PR to remove it (no need for a ticket, we can just refer to this discussion) and there we will make sure all the builds pass.